### PR TITLE
Update docs to include note about existing issues with Async APIs

### PR DESCRIPTION
**Pull request by cheenamalhotra**
Include note about below issues until they get fixed:
- https://github.com/dotnet/SqlClient/issues/593
- https://github.com/dotnet/SqlClient/issues/44

### CODE REVIEW

File: doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml [hunk 1]
```diff
@@ -1698,6 +1698,11 @@ The following example creates a <xref:Microsoft.Data.SqlClient.SqlCommand> and t
 
 ## Remarks
 For more information about asynchronous programming in the .NET Framework Data Provider for SQL Server, see [Asynchronous Programming](/sql/connect/ado-net/asynchronous-programming).
+
+> [!NOTE]
+>  For long running queries on server, you should consider using <xref:Microsoft.Data.SqlClient.SqlCommand.ExecuteNonQuery%2A> due to a known issue with canceling queries with cancellation token. Also, consider canceling execution using the <xref:Microsoft.Data.SqlClient.SqlCommand.Cancel%2A> method.
```
Comment (David-Engel) [inline doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml:1703]:
```suggestion
>  For long running queries on the server, consider using <xref:Microsoft.Data.SqlClient.SqlCommand.ExecuteNonQuery%2A> due to a known issue with canceling queries via a cancellation token. Also, consider canceling execution using the <xref:Microsoft.Data.SqlClient.SqlCommand.Cancel%2A> method.
```

File: doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml [hunk 2]
```diff
@@ -1986,6 +1991,10 @@ The following example creates a <xref:Microsoft.Data.SqlClient.SqlCommand>, and
 
 ## Remarks
 For more information about asynchronous programming in the .NET Framework Data Provider for SQL Server, see [Asynchronous Programming](/sql/connect/ado-net/asynchronous-programming).
+
+> [!NOTE]
+>  For long running queries on server, you should consider using <xref:Microsoft.Data.SqlClient.SqlCommand.ExecuteReader%2A> due to a known issue with canceling queries with cancellation token. Also, consider canceling execution using the <xref:Microsoft.Data.SqlClient.SqlCommand.Cancel%2A> method.
```
Comment (David-Engel) [inline doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml:1996]:
```suggestion
>  For long running queries on the server, consider using <xref:Microsoft.Data.SqlClient.SqlCommand.ExecuteReader%2A> due to a known issue with canceling queries via a cancellation token. Also, consider canceling execution using the <xref:Microsoft.Data.SqlClient.SqlCommand.Cancel%2A> method.
```

File: doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml [hunk 3]
```diff
@@ -2466,6 +2475,10 @@ The following example creates a <xref:Microsoft.Data.SqlClient.SqlCommand> and t
 
 ## Remarks
 For more information about asynchronous programming in the .NET Framework Data Provider for SQL Server, see [Asynchronous Programming](/sql/connect/ado-net/asynchronous-programming).
+
+> [!NOTE]
+>  For long running queries on server, you should consider using <xref:Microsoft.Data.SqlClient.SqlCommand.ExecuteScalar%2A> due to a known issue with canceling queries with cancellation token. Also, consider canceling execution using the <xref:Microsoft.Data.SqlClient.SqlCommand.Cancel%2A> method.
```
Comment (David-Engel) [inline doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml:2480]:
```suggestion
>  For long running queries on the server, consider using <xref:Microsoft.Data.SqlClient.SqlCommand.ExecuteScalar%2A> due to a known issue with canceling queries via a cancellation token. Also, consider canceling execution using the <xref:Microsoft.Data.SqlClient.SqlCommand.Cancel%2A> method.
```

File: doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml [hunk 4]
```diff
@@ -2657,6 +2670,10 @@ The following example creates a <xref:Microsoft.Data.SqlClient.SqlCommand> and t
 ## Remarks
 The **XmlReader** returned by this method does not support asynchronous operations.
 For more information about asynchronous programming in the .NET Framework Data Provider for SQL Server, see [Asynchronous Programming](/sql/connect/ado-net/asynchronous-programming).
+
+> [!NOTE]
+>  For long running queries on server, you should consider using <xref:Microsoft.Data.SqlClient.SqlCommand.ExecuteXmlReader%2A> due to a known issue with canceling queries with cancellation token. Also, consider canceling execution using the <xref:Microsoft.Data.SqlClient.SqlCommand.Cancel%2A> method.
```
Comment (David-Engel) [inline doc/snippets/Microsoft.Data.SqlClient/SqlCommand.xml:2675]:
```suggestion
>  For long running queries on the server, consider using <xref:Microsoft.Data.SqlClient.SqlCommand.ExecuteXmlReader%2A> due to a known issue with canceling queries via a cancellation token. Also, consider canceling execution using the <xref:Microsoft.Data.SqlClient.SqlCommand.Cancel%2A> method.
```

**Comment by DavoudEshtehari** [comment]
Thank you @cheenamalhotra for actively continuing contribution here. I'd rather introducing a new section for SqlClient's known issues under ADO.net doc.

**Comment by David-Engel** [comment]
> I'd rather introducing a new section for SqlClient's known issues under ADO.net doc.

Cheena discussed this with me and I initially proposed the same thing. However, given the impact of these known issues and the fact that there is on quick fix coming, making them apparent in the API docs, where devs are more likely to see them, makes it more likely that they will be able to solve their issues quickly, when they run into them.

**Review by DavoudEshtehari — APPROVED** [review]

**Comment by cheenamalhotra** [comment]
> I'd rather introducing a new section for SqlClient's known issues under ADO.net doc.

You should do that too IMO for lesser popular issues but with significant impact along-with mitigations/solutions - and pull them in API docs when they become more popular and not fixable in short term.

**Review by David-Engel — APPROVED** [review]

**Pull request merged**